daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionCoder.scala

```diff
@@ -197,12 +196,15 @@ object TransactionCoder {
           }
 
         case ne @ NodeExercises(_, _, _, _, _, _, _, _, _, _, _, _, _, _, _) =>
+          def exe = ne
+          def x =
+            EncodeError(node.version, isTooOldFor = "non-empty choice-observers")
           for {
             _ <- Either.cond(
-              test =
-                !((node.version precedes TransactionVersions.minChoiceObservers) && ne.choiceObservers.nonEmpty),
+              test = exe.version >= TransactionVersion.minChoiceObservers ||
+                exe.choiceObservers.isEmpty,
               right = (),
-              left = EncodeError(node.version, isTooOldFor = "non-empty choice-observers")
+              left = x
```

Comment:
Aren't we losing laziness here by binding the error to a variable first? Is that intended? Is this a performance critical code path we should worry about?